Client/Frontend/Browser/LoginsHelper.swift

```diff
@@ -159,32 +175,32 @@ class PasswordHelper: BrowserHelper {
         browser?.addSnackbar(snackBar!)
     }
 
-    private func requestPasswords(password: Password, requestId: String) {
-        profile.passwords.get(QueryOptions(filter: password.hostname), complete: { (cursor) -> Void in
-            var logins = [[String: String]]()
-            for i in 0..<cursor.count {
-                let password = cursor[i]!
-                logins.append(password.toDict())
-            }
+    private func requestLogins(login: LoginData, requestId: String) {
+        profile.logins.getLoginsForProtectionSpace(login.protectionSpace).uponQueue(dispatch_get_main_queue()) { res in
+            var jsonObj = [String: AnyObject]()
+            if let cursor = res.successValue {
+                var logins = [[String: String]]()
+                for i in 0..<cursor.count {
+                    let login = cursor[i]!
+                    logins.append(login.toDict())
```

Comment:
I thought we had `Cursor.map`, but I'm wrong; we only have `.asArray`. But `.asArray` is `map(identity)`, so perhaps we should add it?

You can use `map(cursor, { $0.toDict() })`, though.